### PR TITLE
HOV-11 | Popover component

### PR DESCRIPTION
**Pull request by saurabhsutar192**
Popover Component

## Description
A small popover component that sits around your desired component.

## Motivation and Context
To provide additional information about anything which is very handy and doesn't take any space

## How Has This Been Tested?
Local Build Successful and manual testing

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [X] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: lib/src/components/Popover/Popover.tsx [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+    borderRadius = "4px",
+    width = "fit-content",
+    withArrow = false,
+    arrowSize = "7px",
+    isOpened,
+    onChange = () => {},
+    className,
+    style,
+    zIndex = "1",
+    trapFocus = false,
```
Comment (pruthvi2103) [inline lib/src/components/Popover/Popover.tsx:42]: pl make this a seperate hook too.

File: lib/src/components/Popover/Popover.tsx [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+        isOpened === undefined && setIsOpen(false);
+        break;
+      default:
+        break;
+    }
+  };
+
+  const internalContentKeyDownHandler = (e: KeyboardEvent<HTMLDivElement>) => {
+    const focusableElements =
+      'button, [href], input, select, textarea, [tabindex]:not([tabindex="-1"])';
```
Comment (pruthvi2103) [inline lib/src/components/Popover/Popover.tsx:97]: pls save this in a global const so that we have access to this everywhere

**Review by pruthvi2103 — CHANGES_REQUESTED** [review]
Some comments..

File: lib/src/components/Popover/Popover.tsx (hunk 2 above)
Comment (saurabhsutar192) [reply lib/src/components/Popover/Popover.tsx:97]: sure

File: lib/src/components/Popover/Popover.tsx (hunk 1 above)
Comment (saurabhsutar192) [reply lib/src/components/Popover/Popover.tsx:42]: okay

**Review by pruthvi2103 — APPROVED** [review]
LGTM, just make a `useLockFocus` hook issue @saurabhsutar192

**Pull request merged**